From 64fb9a713737c9b85b30701e7ee3e9c18dc1d7e6 Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Thu, 25 Jan 2024 16:38:56 +0530
Subject: [PATCH] Adding Messege registry entry

-Added Redfish Logs-for Discrete Sensors
-Add OS Critical Stop Sensor Redfish Registry
-Add message registry entry for ACPI system and device

Description:

- Add Discrete Sensors event logs to redfish along with ipmi.
Verified the logs are available in /var/log/redfish & ipmi_sel

- Add "OSCriticalStop" event message entry in RedFish for
Logging events based on the Event in Operating System.

-Add message registry entry for ACPI system and ACPI
 Device discrete Sensor

Tested in Intel Archercity:

1. Verified in RedFish, "OSCriticalStop" events logged properly
GET:
https://<BMC-IP>/redfish/v1/Systems/system/LogServices/EventLog/Entries
{
"@odata.id": "/redfish/v1/Systems/system/LogServices/EventLog/Entries/1694413152",
"@odata.type": "#LogEntry.v1_9_0.LogEntry",
"Created": "2023-09-11T06:19:12+00:00",
"EntryType": "Event",
"Id": "1694413152",
"Message": "Run-time Critical Stop Event is detected in Operating System",
"MessageArgs": [
"Run-time Critical Stop"
],
"MessageId": "OpenBMC.0.1.OSCriticalStop",
"Name": "System Event Log Entry",
"Severity": "OK"
}

Signed-off-by: vipinc <vipinc@ami.com>
---
 .../registries/openbmc_message_registry.hpp   | 60 +++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/redfish-core/include/registries/openbmc_message_registry.hpp b/redfish-core/include/registries/openbmc_message_registry.hpp
index 10fa81ec..5dfb57ef 100644
--- a/redfish-core/include/registries/openbmc_message_registry.hpp
+++ b/redfish-core/include/registries/openbmc_message_registry.hpp
@@ -2010,6 +2010,66 @@ constexpr std::array registry =
             {},
             "None.",
         }},
+    MessageEntry{
+        "SensorProcessorPresence",
+        {
+            "Indicates that Processor Presence is detected",
+            "%1 asserted %2 Event",
+            "OK",
+            2,
+            {"string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "SensorWatchdog2",
+        {
+            "Indicates that Watchdog2 Event Detected",
+            "%1 asserted %2 Event",
+            "OK",
+            2,
+            {"string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "OSCriticalStop",
+        {
+            "Indicates that Critical Event in Operating System is Detected",
+            "%1 Event is detected in Operating System",
+            "OK",
+            1,
+            {"string"},
+            "None.",
+        }},
+    MessageEntry{
+        "SensorSystemACPIPowerState",
+        {
+            "Indicates System ACPI Power State Event Detected",
+            "%1 asserted %2 Event",
+            "OK",
+            2,
+            {"string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "SensorDeviceACPIPowerStateAssert",
+        {
+            "Indicates Device ACPI Power State Event Detected",
+            "%1 asserted %2 Event",
+            "OK",
+            2,
+            {"string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "SensorDeviceACPIPowerStateDeassert",
+        {
+            "Indicates Device ACPI Power State Event Detected",
+            "%1 deasserted %2 Event",
+            "OK",
+            2,
+            {"string", "string"},
+            "None.",
+        }},
     MessageEntry{
         "ResetButtonPressed",
         {
-- 
2.25.1

